File: RxNewDemo/app/src/main/java/ca/six/demo/dagger/whole/core/BaseActivity.java
package ca.six.demo.dagger.whole.core;

import android.app.Activity;

import ca.six.demo.dagger.whole.biz.ActivityComponent;
import ca.six.demo.dagger.whole.biz.DaggerActivityComponent;

public class BaseActivity extends Activity {

    protected ActivityComponent getActivityComponent(){
        WholeApp app = (WholeApp) getApplication();
        return DaggerActivityComponent.builder()
                .appComponent(app.getAppComponent())
                .build();
    }
}

File: Tools/DaggerRxJavaProject/app/src/main/java/ca/six/demo/core/BaseActivity.kt
package ca.six.demo.core

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import ca.six.demo.core.di.PerActivity
import retrofit2.Retrofit
import javax.inject.Inject

@SuppressLint("Registered")
open class BaseActivity : AppCompatActivity() {
    lateinit var retrofit : Retrofit

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val appComponent = (application as BaseApp).appComponent
        val builder = appComponent
                .httpComponentBuilder()
        println("szw $this: [$appComponent,  $builder]") // so this is the problem: builder is not the same
        retrofit = builder
                .baseHttpUrl("https://api.github.com")
                .build()
                .retrofit()
    }
}